### PR TITLE
Add cmake functions to generate Qt targets

### LINKED ISSUE #20288: Create CMake function to add qt-based targets
In order to develop the workbench in parallel with the existing MantidPlot code we need to build all widget code against both Qt4 and Qt5.

To minimise the duplication of cmake code we need to create helper functions to add these targets.

A [prototype](https://github.com/martyngigg/sandbox/blob/master/python/matplotlib/cppwidget/cmake/QtTargetFunctions.cmake) has been created to test this out.

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

This is precursor work so that we can build the Qt-based libraries against both Qt4 & Qt5. The targets are now created with cmake functions to give a consistent naming scheme. This hopefully reduces the clutter in the cmake files.

**To test:**

* Code review
* The build servers should take care of any tests failures.
* It would be a good idea for a test of a package generated against this.

Fixes #20288

**Release Notes** 

*Internal change. Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qtpropertymanager.h [hunk 1]
```diff
@@ -103,7 +103,7 @@ class EXPORT_OPT_MANTIDQT_COMMON QtGroupPropertyManager
     : public QtAbstractPropertyManager {
   Q_OBJECT
 public:
-  QtGroupPropertyManager(QObject *parent = nullptr);
```
Comment (peterfpeterson) [inline qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qtpropertymanager.h]: Please put the `nullptr` back in this file.

File: qt/widgets/common/src/QtPropertyBrowser/qtpropertybrowser.cpp [hunk 2]
```diff
@@ -371,7 +371,7 @@ void QtProperty::setModified(bool modified) {
     \sa insertSubProperty(), removeSubProperty()
 */
 void QtProperty::addSubProperty(QtProperty *property) {
-  QtProperty *after = nullptr;
```
Comment (peterfpeterson) [inline qt/widgets/common/src/QtPropertyBrowser/qtpropertybrowser.cpp]: `nullptr` should come back here too

File: qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qtpropertymanager.h (hunk 1 above)
Comment (martyngigg) [reply qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qtpropertymanager.h]: Thanks for spotting that. I'll put them back.

**Comment by martyngigg** [comment]
I've resurrected those clang-tidy fixes. @samueljackson92 is kindly testing the OSX packaging for me as this is the thing that usually goes wrong.

**Comment by martyngigg** [comment]
It would appear that the OSX package now installs cleanly and starts up without warnings.

**Comment by martyngigg** [comment]
This looks to build everywhere now. @mantidproject/gatekeepers I'd appreciate a quick review of this one to avoid it hovering up conflicts

**Pull request closed without merging**